tests/helpers/test_trigger.py

```diff
@@ -780,3 +790,229 @@ async def good_subscriber(new_triggers: set[str]):
 
     assert trigger_events == [{"sun"}]
     assert "Error while notifying trigger platform listener" in caplog.text
+
+
+async def test_invalid_trigger_platform(
+    hass: HomeAssistant,
+    caplog: pytest.LogCaptureFixture,
+) -> None:
+    """Test invalid trigger platform."""
+    mock_integration(hass, MockModule("test", async_setup=AsyncMock(return_value=True)))
+    mock_platform(hass, "test.trigger", MockPlatform())
+
+    await async_setup_component(hass, "test", {})
+
+    assert "Integration test does not provide trigger support, skipping" in caplog.text
+
+
+async def test_async_track_target_selector_state_change_event_empty_selector(
+    hass: HomeAssistant, caplog: pytest.LogCaptureFixture
+) -> None:
+    """Test async_track_target_selector_state_change_event with empty selector."""
+    calls = []
+
+    @callback
+    def state_change_callback(event):
+        """Handle state change events."""
+        calls.append(event)
+
+    unsub = async_track_target_selector_state_change_event(
+        hass, {}, state_change_callback
+    )
+
+    assert "Target selector {} does not have any selectors defined" in caplog.text
+
+    # Test that no state changes are tracked
+    hass.states.async_set("light.test", "on")
+    await hass.async_block_till_done()
+
+    assert len(calls) == 0
+
+    unsub()
+
+
+async def test_async_track_target_selector_state_change_event(
+    hass: HomeAssistant,
+) -> None:
+    """Test async_track_target_selector_state_change_event with multiple targets."""
+    calls = []
+
+    @callback
+    def state_change_callback(event):
+        """Handle state change events."""
+        calls.append(event)
+
+    async def set_state(entity_id, state):
+        """Set the state of an entity."""
+        hass.states.async_set(entity_id, state)
+        await hass.async_block_till_done()
```

Comment:
Why do we need to await `async_block_till_done`? `state_change_callback` is decorated with `callback`, so I'd expect it to fire immediately?